crates/engine/tree/src/tree/payload_validator.rs

```diff
@@ -522,11 +526,100 @@ where
         // terminate prewarming task with good state output
         handle.terminate_caching(Some(&output.state));
 
+        // Capture parent hash and ancestor overlays for deferred trie input construction.
+        let (parent_hash, overlay_blocks) = ctx
+            .state()
+            .tree_state
+            .blocks_by_hash(block.parent_hash())
+            .unwrap_or_else(|| (block.parent_hash(), Vec::new()));
+
+        // Create a deferred handle to store the sorted trie data.
+        let deferred_trie_data = DeferredTrieData::pending();
+        let deferred_handle_task = deferred_trie_data.clone();
+        let deferred_handle_err = deferred_trie_data.clone();
+        let block_hash = block.hash();
+        let hashed_state_for_trie = hashed_state;
+        let trie_output_for_trie = trie_output;
+        let overlay_blocks_for_trie = overlay_blocks;
+        let deferred_compute_duration =
+            self.metrics.block_validation.deferred_trie_compute_duration.clone();
+
+        // Defer trie sorting and compute trie input to a background task so the validation hot path
+        // can return once the state root is checked. Consumers (DB writes, overlay
+        // providers, proofs) block on the handle only when they actually need the sorted
+        // trie data.
+        let task = move || {
+            let compute_start = Instant::now();
+            let result = std::panic::catch_unwind(AssertUnwindSafe(|| {
```

Comment:
What happens if the task panics and we don't catch it like this? If the whole process panics then I think that's acceptable, nothing in here should be panicking and there's a ton of extra code added here to handle it, imo it'd be better to have simpler code.

Put another way, if something in here is panicking then the logic is so broken it's probably better the process doesn't continue.

Reply:
> but i think thats okay because it likely means that something has been broken before that?

Right, if something panics in here then things have gone very wrong, whatever recovery we do outside of this probably won't actually fix the process.